core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -373,6 +376,41 @@ object DumpLogSegments {
     }
   }
 
+  private class MetadataLogMessageParser extends MessageParser[String, String] {
+    override def parse(record: Record): (Option[String], Option[String]) = {
+      // Per KIP-631, the value format is: int16 (api key), int16 (api version), bytes (encoded ApiMessage)
+      val buf = record.value
+      val apiKey = buf.getShort
+      val apiVersion = buf.getShort
+      val message = MetadataRecordType.fromId(apiKey).newMetadataRecord()
+      val reader = new ByteBufferAccessor(buf)
+      message.read(reader, apiVersion)
+      val maybeJson = apiKey match {
+        // TODO maybe move this switch into MetadataRecordType
```

Comment:
This seems like a good idea. Any reason not to do it now? It would be nice if we didn't have to remember to update this tool after a new message type is introduced.

Reply:
It doesn't _quite_ fit with that class, but I moved it there anyways. @cmccabe might have a better idea where to put it